option to pass custom provision file while (make cluster-up)

/kind enhancement

__Problem:__
 while working behind network proxy, `make cluster-up` fails to setup cluster while downloading CNI config. Also, no `docker pull` or `kubectl apply` is successful as no image gets downloaded due to lack of proxy settings in docker-daemon.

```
.
.
Your Kubernetes master has initialized successfully!

To start using your cluster, you need to run the following as a regular user:

  mkdir -p $HOME/.kube
  sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
  sudo chown $(id -u):$(id -g) $HOME/.kube/config

You should now deploy a pod network to the cluster.
Run "kubectl apply -f [podnetwork].yaml" with one of the options listed at:
  https://kubernetes.io/docs/concepts/cluster-administration/addons/

You can now join any number of machines by running the following on each node
as root:

  kubeadm join 192.168.66.101:6443 --token abcdef.1234567890123456 --discovery-token-ca-cert-hash sha256:23d5e3383df6878d799c4b326f8910d521894dc9a403dcac9dd6a3800236fac3

+ kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/coreos/flannel/v0.9.1/Documentation/kube-flannel.yml
The connection to the server raw.githubusercontent.com was refused - did you specify the right host or port?
provisioning node node01 failed
Makefile:104: recipe for target 'cluster-up' failed
make: *** [cluster-up] Error 1
```
__Expected Behaviour__
- modify provision scripts like `node.sh` or `node<node-no>.sh` etc.
- `gocli run ` should be able to consume custom scripts if indicated explicitly

--- a/README.md
+++ b/README.md
@@ -40,14 +40,14 @@
 * **Deprecated**: `kubevirtci/k8s-1.9.3:`: `sha256:f6ffb23261fb8aa15ed45b8d17e1299e284ea75e1d2814ee6b4ec24ecea6f24b`
 * **Deprecated**: `kubevirtci/k8s-1.10.3:`: `sha256:d6290260e7e6b84419984f12719cf592ccbe327373b8df76aa0481f8ec01d357`
 * **Deprecated**: `kubevirtci/k8s-1.10.4:`: `sha256:2ed70abfa8f6c30d990b76b816577040c0709258cbbd7c70f71a70d547f5544f`
-* `kubevirtci/k8s-1.10.11:`: `sha256:b97e556795a56b9aa1763ddf3a49322b49f96877dccb7a164bbca779df078536`
-* `kubevirtci/k8s-1.11.0:`: `sha256:e02ec414c7673a3644b5ba742b550a124c9195eaacb280151406bf9e8201a95f`
-* `kubevirtci/k8s-1.13.3:`: `sha256:bc0f02d6b970650eb16d12f97e5aa1376b3a13b0ffed6227db98675be2ca1184`
+* `kubevirtci/k8s-1.10.11:`: `sha256:f563a8ab4719e53c2372c4f41dfe55256677ec7afc442dfaebd494926005e3e5`
+* `kubevirtci/k8s-1.11.0:`: `sha256:4d28fb90ac31fad9788fb0316e3bfd268d7ad322bd1b442075183eacacd1ace2`
+* `kubevirtci/k8s-1.13.3:`: `sha256:9c2b78e11c25b3fd0b24b0ed684a112052dff03eee4ca4bdcc4f3168f9a14396`
 * **Deprecated**: `kubevirtci/k8s-multus-1.10.4:`: `sha256:1d16d436347fcb9eba28cad08f6e074d4628e9a097a7325eb1ab87351e7f6d5c`
 * **Deprecated**: `kubevirtci/k8s-multus-1.11.1:`: `sha256:3d35b19105344e270be168920e98287fbefcd5366fdf78681712d05725204559`
-* **Deprecated**: `kubevirtci/k8s-multus-1.12.2:`: `sha256:4974496beb19a30156c125d7721912b54a705926dcbf66c41f570dca286996ba`
-* `kubevirtci/k8s-multus-1.13.3:`: `sha256:d037d12a7c847067b051a627925a344ffcfe191adf5b53dc84ccdabbac510995`
-* `kubevirtci/k8s-genie-1.11.1:`: `sha256:bd62afe346d81f5d3bc0dd3255aaaaa3ae4d91787de2db3ecd94cc22f809b587`
+* **Deprecated**: `kubevirtci/k8s-multus-1.12.2:`: `sha256:d3ebf4a59ed379e59552ce8165596ade121d32e6b8782c192ff4f29863547c78`
+* `kubevirtci/k8s-multus-1.13.3:`: `sha256:75a4433cee49381d1715dcb7fff6599754614d6fc815826a48645621f7fc0ce6`
+* `kubevirtci/k8s-genie-1.11.1:`: `sha256:f1e2e729302bc0f2d4a176807f36c02b44c62605ad80d10249bb85682433b3a4`
 
 ## Using gocli
 

--- a/k8s-genie/scripts/provision.sh
+++ b/k8s-genie/scripts/provision.sh
@@ -49,7 +49,7 @@ yum install --nogpgcheck -y \
     kubeadm-${version} \
     kubelet-${version} \
     kubectl-${version} \
-    kubernetes-cni
+    kubernetes-cni-0.6.0
 
 # Latest docker on CentOS uses systemd for cgroup management
 # kubeadm 1.11 uses a new config method for the kubelet

--- a/k8s-multus/scripts/node01.sh
+++ b/k8s-multus/scripts/node01.sh
@@ -9,8 +9,9 @@ do
 done
 
 kubeadm init --config /etc/kubernetes/kubeadm.conf
+version=`kubectl version --short --client | cut -d":" -f2 |sed  's/ //g' | cut -c2- | cut -d"." -f2`
 
-if [[ ${BASH_REMATCH[1]} -ge "12" ]]; then
+if [[ ${version} -ge "12" ]]; then
 kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f /tmp/flannel-ge-12.yaml
 else
 kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f /tmp/flannel.yaml

--- a/k8s-multus/scripts/provision.sh
+++ b/k8s-multus/scripts/provision.sh
@@ -49,7 +49,7 @@ yum install --nogpgcheck -y \
     kubeadm-${version} \
     kubelet-${version} \
     kubectl-${version} \
-    kubernetes-cni \
+    kubernetes-cni-0.6.0 \
     openvswitch
 
 systemctl start openvswitch

--- a/k8s/scripts/node01.sh
+++ b/k8s/scripts/node01.sh
@@ -11,12 +11,12 @@ done
 kubeadm init --config /etc/kubernetes/kubeadm.conf
 
 version=`kubectl version --short --client | cut -d":" -f2 |sed  's/ //g' | cut -c2- | cut -d"." -f2`
-
-if [[ $version -ge "12" ]]; then
-kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/coreos/flannel/a70459be0084506e4ec919aa1c114638878db11b/Documentation/kube-flannel.yml
+if [[ ${version} -ge "12" ]]; then
+kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f /tmp/flannel-ge-12.yaml
 else
-kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/coreos/flannel/v0.9.1/Documentation/kube-flannel.yml
+kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f /tmp/flannel.yaml
 fi
+
 kubectl --kubeconfig=/etc/kubernetes/admin.conf taint nodes node01 node-role.kubernetes.io/master:NoSchedule-
 
 # Wait for api server to be up.

--- a/k8s/scripts/provision.sh
+++ b/k8s/scripts/provision.sh
@@ -61,15 +61,20 @@ yum install --nogpgcheck -y \
 
 # Latest docker on CentOS uses systemd for cgroup management
 # kubeadm 1.11 uses a new config method for the kubelet
-if [[ $version =~ \.([0-9]+) ]] && [[ ${BASH_REMATCH[1]} -ge "11" ]]; then
+if [[ $version =~ \.([0-9]+) ]] && [[ ${BASH_REMATCH[1]} -ge "12" ]]; then
     # TODO use config file! this is deprecated
     cat <<EOT >/etc/sysconfig/kubelet
 KUBELET_EXTRA_ARGS=--cgroup-driver=systemd --runtime-cgroups=/systemd/system.slice --kubelet-cgroups=/systemd/system.slice --allow-privileged=true --feature-gates="BlockVolume=true,CSIBlockVolume=true,VolumeSnapshotDataSource=true"
 EOT
+elif [[ ${BASH_REMATCH[1]} -ge "11" ]]; then
+    # TODO use config file! this is deprecated
+    cat <<EOT >/etc/sysconfig/kubelet
+KUBELET_EXTRA_ARGS=--cgroup-driver=systemd --runtime-cgroups=/systemd/system.slice --kubelet-cgroups=/systemd/system.slice --allow-privileged=true --feature-gates="BlockVolume=true,CSIBlockVolume=true"
+EOT
 else
     cat <<EOT >>/etc/systemd/system/kubelet.service.d/09-kubeadm.conf
 [Service]
-Environment="KUBELET_EXTRA_ARGS=--cgroup-driver=systemd --runtime-cgroups=/systemd/system.slice --kubelet-cgroups=/systemd/system.slice --allow-privileged=true --feature-gates=BlockVolume=true,CSIBlockVolume=true"
+Environment="KUBELET_EXTRA_ARGS=--cgroup-driver=systemd --runtime-cgroups=/systemd/system.slice --kubelet-cgroups=/systemd/system.slice --allow-privileged=true --feature-gates=BlockVolume=true"
 EOT
 fi
 
@@ -157,14 +162,14 @@ apiServerExtraArgs:
   runtime-config: admissionregistration.k8s.io/v1alpha1
 apiVersion: kubeadm.k8s.io/v1alpha3
 controllerManagerExtraArgs:
-  feature-gates: "BlockVolume=true,,CSIBlockVolume=true,VolumeSnapshotDataSource=true"
+  feature-gates: "BlockVolume=true,CSIBlockVolume=true,VolumeSnapshotDataSource=true"
 kind: ClusterConfiguration
 kubernetesVersion: ${version}
 networking:
   podSubnet: 10.244.0.0/16
 
 EOF
-else
+elif [[ ${BASH_REMATCH[1]} -ge "11" ]]; then
 cat > /etc/kubernetes/kubeadm.conf <<EOF
 apiVersion: kubeadm.k8s.io/v1alpha1
 kind: MasterConfiguration
@@ -175,6 +180,22 @@ apiServerExtraArgs:
   allow-privileged: "true"
 controllerManagerExtraArgs:
   feature-gates: "BlockVolume=true,CSIBlockVolume=true"
+token: abcdef.1234567890123456
+kubernetesVersion: ${version}
+networking:
+  podSubnet: 10.244.0.0/16
+EOF
+else
+cat > /etc/kubernetes/kubeadm.conf <<EOF
+apiVersion: kubeadm.k8s.io/v1alpha1
+kind: MasterConfiguration
+apiServerExtraArgs:
+  runtime-config: admissionregistration.k8s.io/v1alpha1
+  ${admission_flag}: Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,MutatingAdmissionWebhook,ValidatingAdmissionWebhook,ResourceQuota
+  feature-gates: "BlockVolume=true,CustomResourceSubresources=true"
+  allow-privileged: "true"
+controllerManagerExtraArgs:
+  feature-gates: "BlockVolume=true"
 token: abcdef.1234567890123456
 kubernetesVersion: ${version}
 networking:

--- a/k8s/scripts/provision.sh
+++ b/k8s/scripts/provision.sh
@@ -94,9 +94,9 @@ echo br_netfilter >> /etc/modules
 
 kubeadm init --pod-network-cidr=10.244.0.0/16 --kubernetes-version v${version} --token abcdef.1234567890123456
 if [[ ${BASH_REMATCH[1]} -ge "12" ]]; then
-kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/coreos/flannel/a70459be0084506e4ec919aa1c114638878db11b/Documentation/kube-flannel.yml
+    kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f /tmp/flannel-ge-12.yaml
 else
-kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/coreos/flannel/v0.9.1/Documentation/kube-flannel.yml
+    kubectl --kubeconfig=/etc/kubernetes/admin.conf create -f /tmp/flannel.yaml
 fi
 
 # Wait at least for 7 pods

